### PR TITLE
chore: StorageManager unit tests

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
Adds unit tests for util functions. Also fixes a very annoying TS Strict bug. Will follow up with a bug on the JS repo.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ interface UploadFileProps {
   isResumable?: boolean;
   progressCallback: (progress: { loaded: number; total: number }) => void;
   errorCallback: (error: string) => void;
-  completeCallback: (event: { key: string }) => void;
+  completeCallback: (event: { key?: string }) => void;
```
Comment (reesscot) [inline packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts:11]: This is the correct type actually:
https://github.com/aws-amplify/amplify-js/blob/main/packages/storage/src/types/AWSS3Provider.ts#L82

**Review by ioanabrooks — APPROVED** [review]

File: packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts (hunk 1 above)
Comment (calebpollman) [reply packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts:11]: Does that mean `event` is just an empty object when there is no value for `key`? Seems like if there was no value ` key` then `event` itself would just be `undefined` be empty but 🤷  

For improved TS practices, what do you think of typing as:
```suggestionx
  completeCallback: (event: { key: string | undefined }) => void;
```

File: packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts [hunk 2]
```diff
@@ -29,12 +29,18 @@ export function uploadFile({
   if (isResumable === true) {
     return Storage.put(fileName, file, {
       level,
-      resumable: isResumable,
+      resumable: true,
```
Comment (calebpollman) [inline packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts:32]: If this is intentional (which i think it is), can we please add a comment explaining why we default to `true`? Could be confusing to come across this in the future

File: packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts (hunk 1 above)
Comment (reesscot) [reply packages/react/src/components/Storage/StorageManager/utils/uploadFile.ts:11]: Makes sense to me, will update

**Review by calebpollman — APPROVED** [review]

**Pull request merged**